test/unit-test/signaling_api/signaling_api_utest.c

```diff
@@ -75,3 +75,1168 @@ void test_signaling_ConstructDescribeSignalingChannelRequest_BadParams( void )
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate Signaling Construct Describe Channel Request functionality.
+ */
+void test_signaling_ConstructDescribeSignalingChannelRequest_BodyOutofMemory( void )
+{
+    SignalingAwsRegion_t awsRegion = { 0 };
+    SignalingChannelName_t channelName = { 0 };
+    SignalingRequest_t requestBuffer = { 0 };
+    char urlBuffer[ 200 ];
+    char bodyBuffer[ 30 ];         /* The body buffer is small */
+    SignalingResult_t result;
+
+    awsRegion.pAwsRegion = "us-east-1";
+    awsRegion.awsRegionLength = strlen( awsRegion.pAwsRegion );
+
+    channelName.pChannelName = "Test-Channel";
+    channelName.channelNameLength = strlen( channelName.pChannelName );
+
+    requestBuffer.pUrl = urlBuffer;
+    requestBuffer.urlLength = sizeof( urlBuffer );
+    requestBuffer.pBody = bodyBuffer;
+    requestBuffer.bodyLength = sizeof( bodyBuffer );
+
+    result = Signaling_ConstructDescribeSignalingChannelRequest( &( awsRegion ),
+                                                                 &( channelName ), &( requestBuffer ) );
+
+    TEST_ASSERT_EQUAL( SIGNALING_RESULT_OUT_OF_MEMORY,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate Signaling Parse Describe Response fail functionality for Bad Parameters.
+ */
+void test_signaling_ParseDescribeSignalingChannelResponse_BadParams( void )
+{
+    const char message;
+    size_t messageLength = sizeof( message );
+    SignalingChannelInfo_t channelInfo;
+    SignalingResult_t result;
+
+    result = Signaling_ParseDescribeSignalingChannelResponse( NULL,
+                                                              messageLength, &( channelInfo ) );
+
+    TEST_ASSERT_EQUAL( SIGNALING_RESULT_BAD_PARAM,
+                       result );
+
+    result = Signaling_ParseDescribeSignalingChannelResponse( &( message ),
+                                                              messageLength, NULL );
+
+    TEST_ASSERT_EQUAL( SIGNALING_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate Signaling Parse Describe Response functionality for Invalid JSON.
+ */
+void test_signaling_ParseDescribeSignalingChannelResponse_InvalidJson( void )
+{
+    const char * message = "{\"ChannelInfo\": {";        /* Missing Closing Braces */
+    size_t messageLength = strlen( message );
+    SignalingChannelInfo_t channelInfo;
+    SignalingResult_t result;
+
+    result = Signaling_ParseDescribeSignalingChannelResponse( message,
+                                                              messageLength, &( channelInfo ) );
+
+    TEST_ASSERT_EQUAL( SIGNALING_RESULT_INVALID_JSON,
+                       result );
+
+    /* <--------------------------------------------------------------------> */
+
+    const char *message2 = "{"
```

Comment:
Could you leave a comment here to let us know why message2 is invalid JSON message?